plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -839,8 +845,32 @@ public Optional<ConnectorOutputMetadata> finishInsert(ConnectorSession session,
 
         appendFiles.commit();
         transaction.commitTransaction();
+        long newSnapshotId = transaction.table().currentSnapshot().snapshotId();
         transaction = null;
 
+        // TODO (https://github.com/trinodb/trino/issues/15439): it would be good to publish data and stats atomically
```

Comment:
If we move stats collection into Iceberg, we could do this automatically.

Reply:
how can we move stats collection to iceberg?